x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/WatcherLifeCycleService.java

```diff
@@ -88,9 +93,11 @@ public void clusterChanged(ClusterChangedEvent event) {
         }
 
         boolean isWatcherStoppedManually = isWatcherStoppedManually(event.state());
+        boolean isStoppedOrStopping = this.state.get() == WatcherState.STOPPED ||
```

Comment:
this.state.get() could change between those two calls?

Reply:
thanks! addressed in https://github.com/elastic/elasticsearch/pull/43939/commits/eab50a08ab1a73ed16e93324e930741f17efc823